app/js/components/SellerInformation.jsx

```diff
@@ -1,25 +1,44 @@
-import React from 'react';
+import React, {Component} from 'react';
 import PropTypes from 'prop-types';
 import {Row, Col} from 'reactstrap';
 import Blockies from 'react-blockies';
 import Reputation from "./Reputation.jsx";
+import {compactAddress} from '../utils/address';
 
-const SellerInformation = ({address, name, reputation}) => (
-  <Row className="border rounded py-4 m-0 text-center shadow-sm">
-    <Col xs="12">
-      <Blockies seed={address} className="rounded-circle border" scale={8}/>
-    </Col>
-    <Col xs="12">
-      <h4 className="font-weight-bold">{name}</h4>
-    </Col>
-    <Col xs="12">
-      <p className="text-muted">{address}</p>
-    </Col>
-    <Col xs="12">
-      <Reputation reputation={reputation}/>
-    </Col>
-  </Row>
-);
+class SellerInformation extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {addressHovered: false};
+  }
+
+  mouseOverAddress = () => {
+    this.setState({addressHovered: true});
+  };
+
+  mouseOutAddress = () => {
+    this.setState({addressHovered: false});
+  };
+
+  render() {
+    const {address, name, reputation} = this.props;
+    return (<Row className="border rounded py-4 m-0 text-center shadow-sm">
+      <Col xs="12">
+        <Blockies seed={address} className="rounded-circle border" scale={8}/>
+      </Col>
+      <Col xs="12">
+        <h4 className="font-weight-bold">{name}</h4>
+      </Col>
+      <Col xs="12">
+        <p className="text-muted" onMouseOver={this.mouseOverAddress} onMouseOut={this.mouseOutAddress}>
+          {this.state.addressHovered ? address : compactAddress(address)}
```

Comment:
There is an Address component do you think it is possible to use it?

Reply:
Oh right. I'll add the state to that component for the hover

Reply:
Done